server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponse.java

```diff
@@ -34,10 +34,10 @@ public class FieldCapabilitiesIndexResponse extends ActionResponse implements Wr
         this.originVersion = Version.CURRENT;
     }
 
-    FieldCapabilitiesIndexResponse(StreamInput in) throws IOException {
+    FieldCapabilitiesIndexResponse(StreamInput in, IndexFieldCapabilities.Deduplicator fieldDeduplicator) throws IOException {
```

Comment:
Does it need to be provided as a constructor argument? Isn't it always the same type, just created externally? Could we create it within readFields?

Reply:
We can't create the deduplicator inside `readFields` as we need to deduplicate fields between index responses. We can avoid passing the deduplicator in the constructor by deferring the deduplication until we have constructed an index response. However, this requires us to iterate and recreate a list of fields.

Reply:
thanks for explaining :)